charmhelpers/core/host.py

```diff
@@ -756,6 +757,25 @@ def restart_on_change_helper(lambda_f, restart_map, stopstart=False,
     return r
 
 
+def restart_pid_check(service_name, ptable_string=None):
+    """Stop a service, check the processes are gone, start service
+    @param service_name: service name as init system knows it
+    @param ptable_string: string to look for in process table to match service
```

Comment:
This is 'old' style parameter naming; we tend to use this style please:

```python
    :param service_name: the service name as the init system knows it.
    :type service_name: str
    :param ptable_match: the string to lok for in the process table to match the service
    :param ptable_match: str
    :raises: AssertionError if the processes do not go in 'n' seconds
```

Note that I've changed the name of the variable to `ptable_match` as generally it's not normal to encode the type in the name of the variable.

Also, I've added the `Exception` that occurs when the retries are exhausted.  It would also be good to calculate 'n' (which is 3 + 6 + 9 + 12 + 15 seconds ... and I'm not sure about the 15 without looking at the code!)